service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -42,18 +48,93 @@ public DraftCMOService(DateFormatterService dateFormatterService, ObjectMapper m
         this.directionHelperService = directionHelperService;
     }
 
-    public DynamicList getHearingDateDynamicList(CaseDetails caseDetails) {
-        Map<String, Object> data = caseDetails.getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+    public Map<String, Object> extractIndividualCaseManagementOrderObjects(
+        CaseManagementOrder caseManagementOrder,
+        List<Element<HearingBooking>> hearingDetails) {
 
-        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+        if (isNull(caseManagementOrder)) {
+            caseManagementOrder = CaseManagementOrder.builder().build();
+        }
+
+        HashMap<String, Object> data = new HashMap<>();
+        HashMap<String, Object> reviewCaseManagementOrder = new HashMap<>();
+
+        reviewCaseManagementOrder.put("cmoStatus", caseManagementOrder.getCmoStatus());
+
+        data.put("cmoHearingDateList", getHearingDateDynamicList(hearingDetails, caseManagementOrder));
+        data.put("schedule", caseManagementOrder.getSchedule());
+        data.put("recitals", caseManagementOrder.getRecitals());
+        data.put("reviewCaseManagementOrder", reviewCaseManagementOrder);
+
+        return data;
+    }
+
+    public CaseManagementOrder prepareCMO(Map<String, Object> caseData) {
+        DynamicList list = mapper.convertValue(caseData.get("cmoHearingDateList"), DynamicList.class);
+        Map<String, Object> reviewCaseManagementOrder = mapper.convertValue(
+            caseData.get("reviewCaseManagementOrder"), new TypeReference<>() {});
+        CMOStatus cmoStatus = mapper.convertValue(reviewCaseManagementOrder.get("cmoStatus"), CMOStatus.class);
+        Schedule schedule = mapper.convertValue(caseData.get("schedule"), Schedule.class);
+        List<Element<Recital>> recitals = mapper.convertValue(caseData.get("recitals"), new TypeReference<>() {});
+
+        return CaseManagementOrder.builder()
+            .hearingDate(list.getValue().getLabel())
+            .id(list.getValue().getCode())
+            .directions(combineAllDirectionsForCMO(mapper.convertValue(caseData, CaseData.class)))
+            .schedule(schedule)
+            .recitals(recitals)
+            .cmoStatus(cmoStatus)
+            .build();
+    }
+
+    public void prepareCaseDetails(Map<String, Object> caseData, CaseManagementOrder caseManagementOrder) {
```

Comment:
For me that method does two things. It 1) clears some unnecessary keys and 2) copies CMO to another field depending on state. For that reason I'd rather do two methods, named more descriptive. Current method name is super vague and due to the fact it does two things likeliness of reuse is reduced.

Reply:
Agree on the above.